client/src/containers/Dashboard.jsx

```diff
@@ -432,8 +433,7 @@ class Dashboard extends Component {
 
   handleToggleShareProtected(isProtected) {
     this.setState({ passwordAlert: null });
-    const dashboard = getDashboardFromState(this.state.dashboard, true);
-    this.props.dispatch(actions.setShareProtection(dashboard.shareId, { protected: isProtected }));
+    set(this.state, 'dashboard.protected', isProtected);
```

Comment:
this doesnt look right. Maybe:
```
this.setState({ 'dashboard.protected': isProtected });
```

Reply:
or actually:
```
this.setState({ 
  dashboard: {
    ...this.state.dashboard,
    protected: isProtected,
  },
});

Reply:
if you just set it manually (ie not using the `setState` method then the observers wont get triggered